src/Field.js

```diff
@@ -4,26 +4,31 @@ import { FormControl, FormArray, FormGroup } from './model'
 import { isFunction, warning } from './utils'
 
 export default class Field extends React.Component {
-  componentDidMount() {
-    const { control } = this.props
-    // Add listener
-    this.addListener(control)
+  state = {
+    forceUpdate: () => this.forceUpdate()
   }
-  componentWillReceiveProps(nextProps) {
-    const { control } = nextProps
-    if (this.props.control !== control) {
-      this.removeListener(control)
-      this.addListener(control)
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    const { control } = nextProps;
+    if (control !== prevState.lastControl) {
+      Field.removeListener(control);
+      Field.addListener(control, prevState.forceUpdate);
+      return {
+        ...prevState,
+        lastControl: control
```

Comment:
@michal-filip I think it is better to use `componentDidUpdate` so the library will be backward compatible.